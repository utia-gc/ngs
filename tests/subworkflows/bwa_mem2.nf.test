nextflow_workflow {

    name "Test Workflow Bwa_Mem2"
    script "subworkflows/bwa_mem2.nf"
    workflow "Bwa_Mem2"

    test("Bwa_Mem2 builds index and aligns reads -- SE reads.") {
        // instantiate helper objects
        def readsLane1 = new ReadsSELane1()
        def refs       = new RefsEnsembl()

        when {
            params {
                publishDirReports = "${outputDir}/reports"
                publishDirData    = "${outputDir}/data"
            }
            workflow {
                """
                input[0] = [
                    [
                        sampleName:   "${readsLane1.getSampleName()}",
                        sampleNumber: "${readsLane1.getSampleNumber()}",
                        readType:     "${readsLane1.getReadType()}"
                    ],
                    [
                        file("${readsLane1.getR1()}"),
                    ]
                ]
                input[1] = file("${refs.getGenome()}")
                """
            }
        }

        then {
            // test workflow status
            assert workflow.success

            // test SAM
            with(workflow.out.alignments.get(0)) {
                assert size() == 2
                // test SAM file
                assert get(1) ==~ /^.*\/${readsLane1.getSampleName()}\.sam/
            }

            // test reproducibility
            assert snapshot(workflow.out).match()
        }

    }

    test("Bwa_Mem2 builds index and aligns reads -- PE reads.") {
        // instantiate helper objects
        def readsLane1 = new ReadsPELane1()
        def refs       = new RefsEnsembl()

        when {
            params {
                publishDirReports = "${outputDir}/reports"
                publishDirData    = "${outputDir}/data"
            }
            workflow {
                """
                input[0] = [
                    [
                        sampleName:   "${readsLane1.getSampleName()}",
                        sampleNumber: "${readsLane1.getSampleNumber()}",
                        readType:     "${readsLane1.getReadType()}"
                    ],
                    [
                        file("${readsLane1.getR1()}"),
                        file("${readsLane1.getR2()}"),
                    ]
                ]
                input[1] = file("${refs.getGenome()}")
                """
            }
        }

        then {
            // test workflow status
            assert workflow.success

            // test SAM
            with(workflow.out.alignments.get(0)) {
                assert size() == 2
                // test SAM file
                assert get(1) ==~ /^.*\/${readsLane1.getSampleName()}\.sam/
            }

            // test reproducibility
            assert snapshot(workflow.out).match()
        }

    }

}
