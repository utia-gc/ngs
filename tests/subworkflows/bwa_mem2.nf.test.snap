{
    "Bwa_Mem2 builds index and aligns reads -- SE reads.": {
        "content": [
            {
                "0": [
                    [
                        {
                            "sampleName": "SRR1066657",
                            "sampleNumber": "3",
                            "lane": "001",
                            "readType": "single",
                            "rgFields": {
                                "ID": "SRR1066657.001",
                                "SM": "SRR1066657",
                                "LB": "SRR1066657",
                                "PL": "ILLUMINA"
                            }
                        },
                        "SRR1066657_L001.sam:md5,7111ddc8b9797a0e4d04bffaa5c5be34"
                    ]
                ],
                "alignments": [
                    [
                        {
                            "sampleName": "SRR1066657",
                            "sampleNumber": "3",
                            "lane": "001",
                            "readType": "single",
                            "rgFields": {
                                "ID": "SRR1066657.001",
                                "SM": "SRR1066657",
                                "LB": "SRR1066657",
                                "PL": "ILLUMINA"
                            }
                        },
                        "SRR1066657_L001.sam:md5,7111ddc8b9797a0e4d04bffaa5c5be34"
                    ]
                ]
            }
        ],
        "timestamp": "2023-10-30T11:32:27.697108492"
    },
    "Bwa_Mem2 builds index and aligns reads -- PE reads.": {
        "content": [
            {
                "0": [
                    [
                        {
                            "sampleName": "SRR6924569",
                            "sampleNumber": "1",
                            "lane": "001",
                            "readType": "paired",
                            "rgFields": {
                                "ID": "SRR6924569.001",
                                "SM": "SRR6924569",
                                "LB": "SRR6924569",
                                "PL": "ILLUMINA"
                            }
                        },
                        "SRR6924569_L001.sam:md5,1c13b95ac563b45f8d2b7f3789ce8bd4"
                    ]
                ],
                "alignments": [
                    [
                        {
                            "sampleName": "SRR6924569",
                            "sampleNumber": "1",
                            "lane": "001",
                            "readType": "paired",
                            "rgFields": {
                                "ID": "SRR6924569.001",
                                "SM": "SRR6924569",
                                "LB": "SRR6924569",
                                "PL": "ILLUMINA"
                            }
                        },
                        "SRR6924569_L001.sam:md5,1c13b95ac563b45f8d2b7f3789ce8bd4"
                    ]
                ]
            }
        ],
        "timestamp": "2023-10-30T11:32:38.961222639"
    }
}